aws/alarms/athena.tf

```diff
@@ -94,6 +94,98 @@ resource "aws_athena_data_catalog" "dynamodb" {
   type        = "LAMBDA"
 
   parameters = {
-    "function" = data.aws_lambda_function.existing.arn
+    "function"   = data.aws_lambda_function.existing.arn
+    "LambdaRole" = aws_iam_role.athena_dynamodb_role.arn
   }
+}
+
+resource "aws_iam_role_policy" "athena_dynamodb_policy" {
+  name = "athena_dynamodb_policy"
+  role = aws_iam_role.athena_dynamodb_role.id
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        "Effect" : "Allow",
+        "Action" : [
+          "logs:CreateLogGroup",
+          "logs:CreateLogStream",
+          "logs:PutLogEvents"
+        ],
+        "Resource" : "*"
+      },
+      {
+        "Action" : [
+          "glue:GetTableVersions",
+          "glue:GetPartitions",
+          "glue:GetTables",
+          "glue:GetTableVersion",
+          "glue:GetDatabases",
+          "glue:GetTable",
+          "glue:GetPartition",
+          "glue:GetDatabase",
+          "athena:GetQueryExecution",
+          "s3:ListAllMyBuckets"
+        ],
+        "Resource" : "*",
+        "Effect" : "Allow"
+      },
+      {
+        "Action" : [
+          "dynamodb:DescribeTable",
+          "dynamodb:ListSchemas",
+          "dynamodb:ListTables",
+          "dynamodb:Query",
+          "dynamodb:Scan",
+          "dynamodb:PartiQLSelect"
+        ],
+        "Resource" : "${var.dynamodb_audit_logs_arn}",
+        "Effect" : "Allow"
+      },
+      {
+        "Action" : [
+          "kms:Decrypt"
+        ],
+        "Resource" : "*",
```

Comment:
Could you limit this to just the DynamoDB KMS key - we've got this output that will give you the ARN you'll need:
https://github.com/cds-snc/forms-terraform/blob/c6ccabb9f5ff462a95ebb25c1b21369eade29066/aws/kms/outputs.tf#L11-L14